File: src/main/java/me/nepnep/nepbot/LeaveMessage.java
package me.nepnep.nepbot;

import net.dv8tion.jda.api.entities.TextChannel;
import net.dv8tion.jda.api.events.guild.member.GuildMemberRemoveEvent;
import net.dv8tion.jda.api.exceptions.InsufficientPermissionException;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import org.slf4j.LoggerFactory;

import java.util.List;

public class LeaveMessage extends ListenerAdapter {
    @Override
    public void onGuildMemberRemove(GuildMemberRemoveEvent event) {
        List<TextChannel> goodbyeChannels = event.getGuild().getTextChannelsByName("goodbye", false);

        if (goodbyeChannels.size() >= 1) {
            try {
                goodbyeChannels.get(0).sendMessage(String.format("%s Just left the server, goodbye.", event.getUser().getAsTag())).queue();
            } catch (InsufficientPermissionException e) {
                LoggerFactory.getLogger(LeaveMessage.class).debug("Attempted to send leave message but did not have the permissions!");
            }
        }
    }
}

File: src/main/kotlin/me/nepnep/nepbot/LeaveMessage.kt
package me.nepnep.nepbot

import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.events.guild.member.GuildMemberRemoveEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

class LeaveMessage : ListenerAdapter() {
    override fun onGuildMemberRemove(event: GuildMemberRemoveEvent) {
        val guild = event.guild
        val channels = guild.getTextChannelsByName("goodbye", false)

        if (channels.isNotEmpty()) {
            val channel = channels[0]
            if (guild.selfMember.hasPermission(channel, Permission.MESSAGE_SEND)) {
                channel.sendMessage("${event.user.asTag} just left the server, goodbye.").queue()
            }
        }
    }
}